core/src/credential.rs

```diff
@@ -0,0 +1,93 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+use std::{time::{SystemTime}};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use crate::HashMap;
+
+// cred_subject is a generic that implements trait X
+// trait X allows us to encode that object into JSON-LD
+// We provide types that implement trait X for the cred types that we support
+// Users can also user their own types that implement trait X if they need a different structure
+// --- 
+// Default context and Cred types are defaulted but can be redefined 
+
+type VerificationContext = [&'static str; 2];
+
+pub const CONTEXT_CREDENTIALS: VerificationContext = [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+];
+
+pub const CRED_TYPE_PERMANENT_RESIDENT_CARD: &'static str = "PermanentResidentCard";
+pub const CRED_TYPE_BANK_CARD: &'static str = "BankCard";
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct CredentialSubject {
+    id : String,
+    #[serde(flatten)]
+    pub property_set: HashMap<String, Value>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(bound(deserialize = "'de: 'static"))]
+pub struct Credential {
+    #[serde(flatten)]
+    verifiable_credential:VerifiableCredential,
+    proof: IntegrityProof,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(bound(deserialize = "'de: 'static"))]
```

Comment:
Looks like we can't suppress this warning yet:
https://github.com/rust-lang/rust/issues/96956

Once that's merged into main we can add `#[allow(needless_lifetimes)]` here
For now, let's add a comment linking to this post: 
https://users.rust-lang.org/t/restrict-generic-type-to-static-lifetime/16915/4 

The borrow checker currently isn't smart enough to reason around `'de: 'static` bounds.

@LuisOsta @sKudryashov LMK if you have any questions about the two links above